sqlglot/expressions.py

```diff
@@ -266,7 +269,12 @@ def add_comments(self, comments: t.Optional[t.List[str]]) -> None:
         if self.comments is None:
             self.comments = []
         if comments:
-            self.comments.extend(comments)
+            for comment in comments:
+                _, *meta = comment.split(SQLGLOT_META)
+                if meta:
+                    k, v = " ".join(meta).split("=")
+                    self.meta[k.strip()] = v.strip()
```

Comment:
Perhaps we can have an option to omit the `=` and just store a mapping of `k.strip()` -> `True`, to avoid string comparisons later on for keys that act as boolean flags.